lib/jarvis/mixins/pool_delegate.rb

```diff
@@ -0,0 +1,27 @@
+require "jarvis/workpool"
+
+module Jarvis module Mixins module PoolDelegate
+  def pool_delegate(pool_name, &callback)
+    lambda do |request|
+      begin
+        pool = WorkPool.fetch(pool_name)
+        pool.post do
+          # TODO(sissel): Note what command is being executed and at what time.
+          # TODO(sissel): Redirect $stdout and $stderr
+          begin
+            callback.call(request)
+          rescue => e
+            # TODO(sissel): Mark this job as failed
+            request.reply(t("unhandled exception", :exception => e))
+            request.reply(e.backtrace.join("\n"))
+          end
+          # TODO(sissel): Mark this job as complete.
+        end
```

Comment:
Woud you mind extracting this logic to be outside of the thread creation, I find this a bit easier to unit test when we separate logic. Maybe its personal taste.